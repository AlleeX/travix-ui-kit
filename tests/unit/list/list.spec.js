import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import List from '../../../components/list/list';

describe('List', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List items={['London', 'Amsterdam', 'Madrid']} position="vertical" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List items={['London', 'Amsterdam', 'Madrid']} position="horizontal" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should return base class with mods for strings as class and string as mods', () => {
      const wrapper = shallow(
        <List hideBullets items={['London', 'Amsterdam', 'Madrid']} position="vertical" />
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
