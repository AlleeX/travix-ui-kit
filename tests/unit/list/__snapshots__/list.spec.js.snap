exports[`List #render() should return base class with mods for strings as class and string as mods 1`] = `
<ul
  className="ui-list ui-list_position_vertical">
  <li>
    <span
      className="ui-list__content">
      London
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Amsterdam
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Madrid
    </span>
  </li>
</ul>
`;

exports[`List #render() should return base class with mods for strings as class and string as mods 2`] = `
<ul
  className="ui-list ui-list_position_horizontal">
  <li>
    <span
      className="ui-list__content">
      London
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Amsterdam
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Madrid
    </span>
  </li>
</ul>
`;

exports[`List #render() should return base class with mods for strings as class and string as mods 3`] = `
<ul
  className="ui-list ui-list_position_vertical ui-list_no_bullets">
  <li>
    <span
      className="ui-list__content">
      London
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Amsterdam
    </span>
  </li>
  <li>
    <span
      className="ui-list__content">
      Madrid
    </span>
  </li>
</ul>
`;
